pkg/osbuild/rpm_stage_test.go

```diff
@@ -180,3 +180,96 @@ func TestNewRpmStageSourceFilesInputs(t *testing.T) {
 		}
 	}
 }
+
+func TestInvalidHashAlgorithmForNewRpmStageSourceFilesInputs(t *testing.T) {
+
+	assert := assert.New(t)
+	require := require.New(t)
+
+	pkgSpecs := []rpmmd.PackageSpec{
+		{
+			Name:           "openssl-libs",
+			Epoch:          1,
+			Version:        "3.0.1",
+			Release:        "5.el9",
+			Arch:           "x86_64",
+			RemoteLocation: "https://example.com/repo/Packages/openssl-libs-3.0.1-5.el9.x86_64.rpm",
+			Checksum:       "invalid:fcf2515ec9115551c99d552da721803ecbca23b7ae5a974309975000e8bef666",
```

Comment:
@achilleas-k  I was expecting this to fail, however it's passing.

The `invalid:` prefix is the only difference between this test and `TestNewRpmStageSourceFilesInputs` above it.

Reply:
We don't verify the checksum string in osbuild-composer or images.  This would have to be run against the stage in osbuild to fail.  If we want to have that verification early, we can add it as a `validate()` function that gets called as part of `New...()`.

We do have a `validate()` function for file references: https://github.com/osbuild/images/blob/41bfe61e14d32c7b176c86e79b4d37c53cd6fe23/pkg/osbuild/files_input.go#L309-L313
but it only checks if a checksum is included.

We can expand the validate to check for the algorithm strings that are supported by osbuild.  That function panics if it's invalid, so the test would have to be adapted to catch the panic instead.

Reply:
> We can expand the validate to check for the algorithm strings that are supported by osbuild. That function panics if it's invalid, so the test would have to be adapted to catch the panic instead.

That is the original intention, I didn't modify the test initially in order to see where exactly it will fail.

Reply:
Adding stricter checks to the checksum validator.